src/Orleans.Core/Core/DefaultClientServices.cs

```diff
@@ -17,6 +18,9 @@ public static void AddDefaultServices(IClientBuilder builder, IServiceCollection
         {
             services.TryAddSingleton<TelemetryManager>();
             services.TryAddFromExisting<ITelemetryProducer, TelemetryManager>();
+            // if no IHostEnvironmentStatistics registered, use the dummy one
+            services.TryAddSingleton<IHostEnvironmentStatistics, DummyHostEnvironmentStatistics>();
```

Comment:
I don't think we should inject a dummy one y default. If user missed the warning line of   "No implementation of IHostEnvironmentStatistics was found. Load shedding will not work yet", user may actually take those dummy statistic reading seriously.

Reply:
I'm fine with injecting a NoOp one, to make dependency injection easier. But you may need to change `IHostEnvironmentStatistics ` interface to have a no-op implementation.

Reply:
Unless the user is doing something very custom, *I believe* the default way of using the silo builder is that `AddDefaultServices` will be called before all app-level services are configured. You might want to double check, but that's what I remember, so the comment would not be accurate.
IIRC, the reason many of these use the `TryAddXXX` methods is to allow infrastructure to be tweaked, but not necessarily for end users to provide their own by default.